ui/admin-portal/src/app/components/account/user-change-password/user-change-password.component.spec.ts

```diff
@@ -0,0 +1,121 @@
+import {ComponentFixture, fakeAsync, TestBed, tick, waitForAsync} from '@angular/core/testing';
+import { UserChangePasswordComponent } from './user-change-password.component';
+import { FormBuilder, ReactiveFormsModule } from '@angular/forms';
+import { ActivatedRoute, Router } from '@angular/router';
+import { of, throwError } from 'rxjs';
+import { UserService } from '../../../services/user.service';
+
+fdescribe('UserChangePasswordComponent', () => {
+  let component: UserChangePasswordComponent;
+  let fixture: ComponentFixture<UserChangePasswordComponent>;
+  let mockUserService: jasmine.SpyObj<UserService>;
+  let mockActivatedRoute: any;
+  let mockRouter: any;
+
+  beforeEach(waitForAsync(() => {
+    mockUserService = jasmine.createSpyObj('UserService', ['checkPasswordResetToken', 'resetPassword']);
+
+    mockActivatedRoute = {
+      paramMap: of({ get: () => 'mockToken' })
+    };
+    mockRouter = jasmine.createSpyObj('Router', ['navigate']);
+     // Mock implementation for checkPasswordResetToken
+    mockUserService.checkPasswordResetToken.and.returnValue(of({ success: true }));
+// Mock implementation for resetPassword
+    mockUserService.resetPassword.and.returnValue(of({ success: true }));
+    TestBed.configureTestingModule({
+      declarations: [UserChangePasswordComponent],
+      imports: [ReactiveFormsModule],
+      providers: [
+        FormBuilder,
+        { provide: UserService, useValue: mockUserService },
+        { provide: ActivatedRoute, useValue: mockActivatedRoute },
+        { provide: Router, useValue: mockRouter }
+      ]
+    })
+    .compileComponents();
+
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(UserChangePasswordComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should initialize form based on token', () => {
+    expect(component.changePasswordForm).toBeDefined();
+    expect(component.token).toBe('mockToken');
+  });
+  //
+  it('should initialize form without token', () => {
+    mockActivatedRoute.paramMap = of({ get: () => null });
+    fixture = TestBed.createComponent(UserChangePasswordComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+    expect(component.changePasswordForm).toBeDefined();
+    expect(component.token).toBeNull();
+  });
+
+  it('should update password successfully', fakeAsync(() => {
+    component.reset = true;
+    component.changePasswordForm.setValue({
+      token: 'mockToken',
+      password: 'newPassword',
+      passwordConfirmation: 'newPassword'
+    });
+    mockUserService.resetPassword.and.returnValue(of(null));
+    component.updatePassword();
+    tick(2100);
+     expect(mockUserService.resetPassword).toHaveBeenCalledWith({
+      token: 'mockToken',
+      password: 'newPassword',
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

We detected credentials hardcoded in your code, which might allow unauthorized users to access your account. We recommend you to store your credentials outside of the code in a configuration file, a database, or a management service for secrets and retrieve from it. For more information, see [CWE-259](https://cwe.mitre.org/data/definitions/259.html) and [CWE-798](https://cwe.mitre.org/data/definitions/798.html).



Suggested remediation:
Do not hardcode password or access key. Instead, try to fetch it from environment to deny unauthorized users to access your account.
```
@@ -75,3 +75,3 @@
       token: &#39;mockToken&#39;,
-      password: &#39;newPassword&#39;,
+      password: process.env.ENV_VARIABLE,
       passwordConfirmation: &#39;newPassword&#39;

```